### PR TITLE
Signup: Implement a 'crowdsignal' signup flow based on 'wpcc'

### PR DESCRIPTION
**Pull request by ice9js**
#### Changes proposed in this Pull Request

This patch adds a new `crowdsignal` signup flow which is based on the existing WPCC flow. The changes for now include:

- Enabled social signup (Google)
- Replaced the username field with first and last names

The main goal we hope to achieve with these changes is to simplify the signup flow for new Crowdsignal users. ( pabtAt-3v-p2, pabtAt-3J-p2 )

**Requires D20467-code.**

screen shot 2018-11-07 at 23 48 44

#### Testing instructions

- Go to crowdsignal.com and click 'Create survey' and wait for the redirect to WordPress.com to happen.
- Change this part of the url: `/start/wpcc` to `/start/crowdsignal` and update the domain name according to your testing environment.
- You should see the signup form with first and last name fields, as well as the username field missing. Social signup should be enabled.
- Ensure appropriate errors are communicated for each field when filled in incorrectly. The names can only contain letters.
- Submit the form once it's filled in correctly - you should be signed up with a new account.

Also ensure social signup is working as expected.

### CODE REVIEW

File: client/blocks/signup-form/index.jsx [hunk 1]
```diff
@@ -185,6 +188,16 @@ class SignupForm extends Component {
 		}
 	}
 
+	sanitizeName( name ) {
```
Comment (dbtlr) [inline client/blocks/signup-form/index.jsx]: This whole function is all sorts of wrong for internationalization reasons ...

Is there a reason we need to do this at all? If someone want's to say their name is: `J4min B*tz`, who's to say that it isn't? :)

I'd use a more Unicode example with Ás and such, but hopefully, you get the point. We can't predict what characters are in people's names. :)

File: client/blocks/signup-form/index.jsx [hunk 2]
```diff
@@ -194,19 +207,37 @@ class SignupForm extends Component {
...
+				password: fields.password,
 			} );
 		}
 	};
 
 	validate = ( fields, onComplete ) => {
-		wpcom.undocumented().validateNewUser( fields, ( error, response ) => {
+		const data = pick( fields, [ 'email', 'username', 'password' ] );
+
+		if ( this.props.useFullName ) {
```
Comment (dbtlr) [inline client/blocks/signup-form/index.jsx]: Is there a case where you may want a full name AND username?

I'd suggest using 2 explicit props for turning these on and off. :)

File: client/blocks/signup-form/index.jsx [hunk 3]
```diff
@@ -262,6 +293,16 @@ class SignupForm extends Component {
 				}
 			}
 
+			if ( messages.first_name ) {
```
Comment (dbtlr) [inline client/blocks/signup-form/index.jsx]: Why use the snake_case notation at all? Seems a hassle to have to convert things.

File: client/blocks/signup-form/index.jsx [hunk 4]
```diff
@@ -391,6 +432,10 @@ class SignupForm extends Component {
 			username: formState.getFieldValue( this.state.form, 'username' ),
 			password: formState.getFieldValue( this.state.form, 'password' ),
 			email: formState.getFieldValue( this.state.form, 'email' ),
+			extra: {
+				first_name: formState.getFieldValue( this.state.form, 'firstName' ),
```
Comment (dbtlr) [inline client/blocks/signup-form/index.jsx]: You don't have `user_name`, so maybe just use `firstname` and `lastname` here.

Comment (ice9js) [reply client/blocks/signup-form/index.jsx]: This is the data that's passed as it to the API request which expects snake_case keys.  
Doing this automagically higher up might be confusing unless we also put in the effort to update some of the wrapper components that add properties manually.

File: client/blocks/signup-form/index.jsx (hunk 3 above)
Comment (ice9js) [reply client/blocks/signup-form/index.jsx]: This is the 'raw' data from the API hence the need to convert.  
I cleaned it up a little by using `mapKeys` and `camelCase` when it's loaded.

File: client/blocks/signup-form/index.jsx (hunk 2 above)
Comment (ice9js) [reply client/blocks/signup-form/index.jsx]: I broke up `useFullName` into `displayNameInput` and `displayUsernameInput`.

**Pull request merged**